enterprise/cypher/spec-suite-tools/src/test/scala/cypher/features/Neo4jExceptionToExecutionFailed.scala

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package cypher.features
+
+import org.neo4j.kernel.api.exceptions.Status
+import org.opencypher.tools.tck.api.ExecutionFailed
+
+object Phase {
+  val runtime = "runtime"
+  val compile = "compile time"
+}
+
+object Neo4jExceptionToExecutionFailed {
+
+  def convert(phase: String, t: Throwable): ExecutionFailed = {
+    val neo4jException = t match {
+      case re: RuntimeException => re
+      case _ => throw t
+    }
+    val errorType = Status.statusCodeOf(neo4jException)
+    val msg = neo4jException.getMessage
+    val detail = phase match {
+      case Phase.compile => compileTimeDetail(msg)
+      case Phase.runtime => runtimeDetail(msg)
+    }
+    ExecutionFailed(errorType.toString, phase, detail)
```

Comment:
Since ExecutionFailed expects a String as the third argument, what is the reason we convert all the Neo4J exception messages to less verbose Strings?